astro/src/content/quickstarts/quickstart-javascript-nextjs-web.mdx

```diff
@@ -0,0 +1,283 @@
+```
+
+Also copy an image file into a new directory within `public` called `img`.
+
+```shell
+mkdir ./public/img && cp ../complete-application/public/img/money.jpg ./public/img/money.jpg
+```
+
+As we will be recreating all of the files in our app directory, please delete all files within `/src/app`.
+
+```shell
+rm -rf ./src/app && mkdir ./src/app
+```
+
+#### Authentication API
+
+Next.js 13.2 introduced <a href="https://nextjs.org/docs/app/building-your-application/routing/router-handlers" target="_blank">Route Handlers</a>, which are the preferred way to handle REST-like requests. In the `Changebank` application we can configure NextAuth.js FusionAuth's provider in a new route handler by creating a file within `src/app/api/auth/[...nextauth]/route.ts`.
+
+On first load of Next.js this file will make sure that we have all of the correct environment variables. The variables are then exported in an object called `authOptions` which can be imported on the server when we need to get our session using `getServerSession`.
+
+The `FusionAuthProvider` is then provided to `NextAuth` as a provider for any `GET` or `POST` commands that are sent to the `/api/auth/*` route.
+
+`src/app/api/auth/[...nextauth]/route.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/api/auth/[...nextauth]/route.ts"}
+  lang="ts" />
+
+#### Styles
+
+Create a new file `src/app/globals.css` copy the below CSS for the changebank application.
+
+`src/app/globals.css`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/globals.css"}
+  lang="ts" />
+
+#### Login Button
+
+Create a new file in ``src/components/login-button.tsx` that will be used for a button component. Our login button will only be used on the client side so make sure to add `use client` at the top of this file.
+For this button we can use the `signIn` and `signOut` functions from `next-auth/react`. By passing in the session from our pages we can determine if the `Log in` or `Log out` should be shown. (Yeah I know if we could have all decided on Log in or Sign in over the last 20 years that would have been great!)
+
+`src/components/login-button.tsx`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/components/login-button.tsx"}
+  lang="ts" />
+
+#### Layout
+
+<Aside type="tip">
+If this is your first time using Next.js's App Router, I would highly suggest reading through at least the <a href="https://nextjs.org/docs/app/building-your-application/routing" target="_blank">Routing Fundamentals</a>
+</Aside>
+
+Below you will find the full code for the [Root Layout](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required)
+
+This has the overall structure of our application, the other pages will be added where `{children}` is located.
+
+There are two items of note:
+1. Getting the session information - `const session = await getServerSession(authOptions);`
```

Comment:
It maybe because I am new, but this is getServerSession part of next-auth or nextjs?  Should clarify for simpletons like me.